dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -347,6 +358,130 @@ public void delete(Context context, Relationship relationship, boolean copyToLef
         }
     }
 
+
+    /**
+     * Utility method to ensure discovery is updated for the 2 items
+     * This method is used when creating, modifying or deleting a relationship
+     * The virtual metadata of the 2 items may need to be updated, so they should be re-indexed
+     *
+     * @param context           The relevant DSpace context
+     * @param relationship      The relationship which has been created, updated or deleted
+     * @throws SQLException     If something goes wrong
+     */
+    private void updateItemsInRelationship(Context context, Relationship relationship) throws SQLException {
+        // Since this call is performed after creating, updating or deleting the relationships, the permissions have
+        // already been verified. The following updateItem calls can however call the
+        // ItemService.update() functions which would fail if the user doesn't have permission on both items.
+        // Since we allow this edits to happen under these circumstances, we need to turn off the
+        // authorization system here so that this failure doesn't happen when the items need to be update
+        context.turnOffAuthorisationSystem();
+        try {
+            // Set a limit on the total amount of items to update at once during a relationship change
+            int max = configurationService.getIntProperty("relationship.update.relateditems.max", 20);
+            // Set a limit on the total depth of relationships to traverse during a relationship change
+            int maxDepth = configurationService.getIntProperty("relationship.update.relateditems.maxdepth", 5);
```

Comment:
These two configs probably should be added to a config file (maybe a new `relationships.cfg`?) so that they may be described.  It looks (to me) like these settings need clarification, as both the `max` and the `maxdepth` seem to define how many items are updated via each of the left and right item relationships (and the both seem to exclude the immediate left & right item)...

So, if I'm understanding correctly:
* `max` defines the maximum number of items to be updated for both the left and right items in the relationship.  So, a value of `20` means a max of 20 on the left side and a max of 20 on the right side (so a max of 40 total).
* `maxdepth` is similar except specific to the depth...so a value of `5` means that a maximum of 5 levels deep will be scanned for updates on both the left side and the right side.

Reply:
the config has been created, the max is not separate for left and right, so it deviates a bit from your description